### PR TITLE
Make configure compatible with PHP 7.4

### PR DESCRIPTION
**Pull request by ausi**
This is a followup to #29 using separate config files [as requested](https://github.com/travis-ci/php-src-builder/pull/29#issuecomment-472557912) by @BanzaiMan

### CODE REVIEW

**Comment by ausi** [comment]
The libonig issue still needs to be resolved I think.  

https://github.com/travis-ci/php-src-builder/pull/29#issuecomment-462346556

> the build is now likely to fail due to a missing libonig dependency.
> 
> @BanzaiMan To get an additional dependency, would it be sufficient to install `libonig-dev` in `.travis.yml`, or would this result in a missing shared object dependency when the built binary is used? That is, does something additional need to be done to make sure that the necessary SO is also present in the container/VM in some manner?

**Review by joshk — COMMENTED** [review]
Due to the use of : 

```
$TRAVIS_BUILD_DIR/default_configure_options.${RELEASE}
```
in the install files, and the use of `default_configure_options.trusty-master` style configure files, it might be best to use the `custom_configure_options` file and then add them to the default file used.

**Comment by nikic** [comment]
Why are you disabling PEAR on master? It should be compatible with PHP 8. At least the Console_GetOpt issue you're referencing was fixed a few weeks ago.

**Comment by joshk** [comment]
hey @nikic, I did lots of testing but it seems as though the version of Console_Getopt packaged in PEAR is old and does not include the `each()` fix.

File: default_configure_options.bionic [hunk 1]
```diff
@@ -0,0 +1,49 @@
+--enable-intl
```
Comment (nikic) [inline default_configure_options.bionic:1]: Doesn't this one also need all the version specific variations, at least for baseline, 7.4, master? (Which is also the reason why I still think that #29 is a more scalable solution to this problem.)

**Comment by nikic** [comment]
@joshk Just tested this locally and you are absolutely right. This must be a recent regression as this was working a few days ago :/ I'll try to get this fixed.

File: default_configure_options.bionic (hunk 1 above)
Comment (joshk) [reply default_configure_options.bionic:1]: sorry, i borrowed this from the xenial base line default for my testing.

**Comment by joshk** [comment]
awesome, thanks @nikic 
I spent a lot of time hunting it down and then seeing if I could build an updated release locally, but since PEAR needs PEAR to build PEAR, I was a bit stuck.

**Comment by joshk** [comment]
I have to admit that I hijacked this PR a little to test some build system improvements. Once this gets merged in I can create a PR with my tweaks.
(this will make it easier to build other OS versions)

File: bin/compile-extension-memcache [hunk 2]
```diff
@@ -7,7 +7,7 @@ travis_time_start
 
 if [[ ! $VERSION =~ ^7 && ! $VERSION =~ ^master$ ]]; then
 	pecl download memcache-beta
-	tar zxvf memcache*.tgz && pushd memcache*
+	tar zxvf memcache*.tgz && pushd memcache*/
```
Comment (BanzaiMan) [inline bin/compile-extension-memcache:10]: What are these trailing slashes for?

**Comment by nikic** [comment]
Forgot to report back here, the go-pear.phar issue has been resolved (again...) so it should be fine to also install on master.

**Comment by nikic** [comment]
Also in the meantime `--enable-maintainer-zts` has become `--enable-zts` on master.

**Pull request merged**